pkg/core/storage/memcached_store.go

```diff
@@ -77,21 +83,120 @@ func (s *MemCachedStore) GetBatch() *MemBatch {
 
 // Seek implements the Store interface.
 func (s *MemCachedStore) Seek(key []byte, f func(k, v []byte)) {
+	// Create memory store snapshot not to hold the lock.
+	memSnapshot := NewMemoryStore()
```

Comment:
Waaay expensive and I'm not sure really needed. You have to walk through entire `mem` and `del` anyway, but you already have a `key` here, so you can do that and grab only the items you actually need, add them to the slice and then just sort this slice.